SimPEG/electromagnetics/frequency_domain/fields.py

```diff
@@ -827,13 +836,16 @@ def _j(self, bSolution, source_list):
         :return: primary current density
         """
 
-        j = self._edgeCurl.T * (self._MfMui * bSolution)
+        if self.simulation.permittivity is None:
+            j = self._edgeCurl.T * (self._MfMui * bSolution)
 
-        for i, src in enumerate(source_list):
-            s_e = src.s_e(self.simulation)
-            j[:, i] = j[:, i] - s_e
+            for i, src in enumerate(source_list):
+                s_e = src.s_e(self.simulation)
+                j[:, i] = j[:, i] - s_e
 
-        return self._MeI * j
+            return self._MeI * j
+        else:
+            return self._MeI * self._MeSigma * self._e(bSolution, source_list)
```

Comment:
```suggestion
            return self._MeI * (self._MeSigma * self._e(bSolution, source_list))
```
avoid the matrix-matrix multiplication